debian_script.bsh

```diff
@@ -8,20 +8,16 @@ SRC_DIR=${SRC_DIR:-/src}
 REPO_CODENAME=${REPO_CODENAME:-$(source /etc/os-release; echo $VERSION | sed -r 's|.*\((.*)\)|\1|')}
 
 mkdir -p ${GIT_LFS_BUILD_DIR}
-cp -r -T "${SRC_DIR}" "${GIT_LFS_BUILD_DIR}/i386"
-cd "${GIT_LFS_BUILD_DIR}/i386"
-git clean -xdf .
-git checkout-index --force --all
-dpkg-buildpackage -d -us -uc -b -ai386
 
-cp -r -T "${SRC_DIR}" "${GIT_LFS_BUILD_DIR}/amd64"
-cd "${GIT_LFS_BUILD_DIR}/amd64"
-git clean -xdf .
-git checkout-index --force --all
-dpkg-buildpackage -d -us -uc -b
+for ARCH in i386 amd64 arm arm64; do
```

Comment:
I don't think we want to use the architecture `arm`.  That's the obsolete old ABI architecture, not the new EABI architecture.  We'd want `armel`, I think.  We'd also need to be sure that we tell Go to build for a suitable version of the processor with soft float, which probably means setting `GOARM=5` in that case.

Also, I have concerns about whether this will build properly on an `amd64` system.  We're switching to using GitHub Actions for our release process, so we don't have control over what hardware builds our packages.  I think we may run the testsuite during packaging, which isn't likely to work for foreign architectures.  Have you tested this?  You can do so by running the `build_dockers.bsh` script in this repo and then running `DOCKER_AUTOPULL=0 docker/run_dockers.bsh` in git-lfs/git-lfs.

Reply:
Hey @bk2204,

yes thanks for pointing this out. We could also use `GOARM=7` for floating values support. This will be supported by raspberry pi's 2 or higher. If older stuff needs to be supported we could also go with `GOARM=5`.

regarding your concerns about building it on amd64. The good part is that docker is involved. In General, it's possible to run different architectures on an amd64 Linux when [binfmt_misc](https://github.com/torvalds/linux/blob/master/Documentation/admin-guide/binfmt-misc.rst) is enabled on the kernel. So then you can register the correct header of architecture to the matching qemu-user-static binary. Everything will be emulated and can be run on amd64.  You don't need to be afraid there is all ready a docker container that does the job of registery every binary for the static. 

Check [multiarch/qemu-user-static/](https://hub.docker.com/r/multiarch/qemu-user-static/) There you can see the instructions. I think this would be possible in ci. In General we need to ensure that the correct image with targeted architectures binaries is fetched and started.

I used this way a lot of times. This is also directly out of the box by docker-for-mac. So you can try it out if you want.

Reply:
According to [the Go wiki](https://github.com/github/babeld/pull/833), we need to use `GOARM=5` for `armel` and `GOARM=6` if we wanted `armhf`.  Those are the versions we'd need to specify for appropriate compatibility with the Debian architectures.  It doesn't look like the `dh-golang` package provides this for us, so we'll likely need to specify it ourselves here.

I'm not opposed to seeing a PR against the git-lfs/git-lfs repository's release Actions workflow  to build binaries using the appropriate architecture.  Since we do the builds on an Ubuntu system, we could just install using the`qemu-user-static` package instead of needing a privileged Docker container.  However, since we're doing a release soon (probably next week), I'd want to see both this PR and a PR there ready before merging so that we don't leave the tree in a broken state.

Reply:
Some updates from my side. I updated everything to match the correct architecture. Regarding the qemu-stuff, we should discuss this in #3846.